File: models/inventory/InventoryItem.js
import mongoose from 'mongoose';

const inventoryItemSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        unique: true
    },
    quantity: {
        type: Number,
        required: true,
        default: 0
    },
    unit: {
        type: String,
        required: true
    },
    supplier: {
        type: String
    }
}, { timestamps: true });

export default mongoose.model('InventoryItem', inventoryItemSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/InventoryItem.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "inventory_items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class InventoryItem extends BaseEntity {

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false)
    private Integer quantity = 0;

    @Column(nullable = false)
    private String unit;

    private String supplier;
}
